### ISSUE TITLE
GQL -  Error: Interface field Craft_FormInterface.fields expected but Craft_Form does not provide it.

### ISSUE THREAD

**Issue by danieladarve** [verbb/formie#343]
**Description**
Getting an error using the Craft Gatsby Plugin after upgrading to the latest version of Formie.
`` Error: Interface field Craft_FormInterface.fields expected but Craft_Form does not provide it.``

The formie field is inside a Super Table field inside a Matrix.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: No
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.3.16.1
- Craft version: Craft CMS 3.6.5.1 Pro
- Multi-site: No
- Super table version: 2.6.7

**Additional context**

**Comment by engram-design** [comment]
When does this error occur? Can you provide the query you're using?

Does this also happen when the Formie field isn't inside a Matrix or Super Table field?

**Comment by danieladarve** [comment]
It happens when I try to source the plug-in using the source craft’s Gatsby
plug-in. But I now believe it is an issue on the Gatsby plug-in side of
things and their recent update.

It happens in all scenarios, super table field, matrix + super table,
standalone formie field.

**Comment by engram-design** [comment]
Cool, I'm having my own issues spinning up a Craft + Gatsby site (using the [starter-blog](https://github.com/craftcms/starter-blog) setup), so struggling to get a working setup to test this out. I imagine just when running `gatsby develop` or `npm run develop` (whatever your setup) and it does its compiling thing, it's thrown during that? I just wanted to check if you were using any queries in your js files that's worth setting up on my end, or if that just isn't applicable.

I'll keep looking into it.

**Comment by danieladarve** [comment]
yea thats right, thats when the error comes up, at startup when you run
gatsby develop when formie is sourced from craft.

**Comment by danieladarve** [comment]
this is my gatsby-config.js

```
module.exports = {
  siteMetadata: {
    title: `Site Name`,
    description: `Description`,
    author: `Author`,
    siteURL: frontEndUrl(NODE_ENV),
  },
  plugins: [
    `gatsby-plugin-postcss`,
    {
      resolve: `gatsby-source-craft`,
      options: {
        fragmentsDir: `${__dirname}/craft-fragments/`,
        debugDir: `${__dirname}/craft-debug/`,
        looseInterfaces: false,
        sourcingParams: {
          images_Asset: {
            kind: '"image"',
          },
        },
      },
    },
    // this (optional) plugin enables Progressive Web App + Offline functionality
    // To learn more, visit: https://gatsby.dev/offline
    // `gatsby-plugin-offline`,
  ],
};
```

the queries that I have simply query the matrix blocks and the fields in them.. so for instance for the homepage i have something like:
```
import React, { useContext } from "react";
import { graphql } from "gatsby";
import PropTypes from "prop-types";
import PageBuilder from "../components/PageBuilder";
import Seo from "../components/seo";

export const query = graphql`
  query {
    entry: craftHomeHomeEntry {
      id
      title
      pageBlockContent {
        ...PageBlocks
      }
    }
  }
`;

const PageHome = ({ pageContext, location, data }) => (
  <>
    <Seo seo={data.entry.seo} title={data.entry.title} location={location} />
    <PageBuilder
      blocks={data.entry.pageBlockContent}
      layout={pageContext.layoutType}
    />
  </>
);

PageHome.propTypes = {
  pageContext: PropTypes.object,
  location: PropTypes.object,
  data: PropTypes.object,
};

export default PageHome;
```

**Comment by engram-design** [comment]
Any further development with this one - still an issue on the latest version?

**Comment by danieladarve** [comment]
Hi Josh,
I will check tonight and let you know.

Thanks

**Comment by danieladarve** [comment]
@joshuabaker Using the craft's Gatsby Helper@1.0.1 and gatsby-source-craft@1.0.0 for Gatsby v2 and the lastest version of CraftCMS

I now get:
 ```
ERROR #11321  PLUGIN

"gatsby-source-craft" threw an error while running the sourceNodes lifecycle:

Failed to execute query LIST_Form.


  468 |     if (remoteConfigVersion !== localConfigVersion || !localContentUpdateTime) {
  469 |         reporter.info("Cached content is unavailable or outdated, sourcing _all_ nodes.");
> 470 |         await sourceAllNodes(config);
      |         ^
  471 |     }
  472 |     else {
  473 |         reporter.info(`Craft config version has not changed since last sourcing. Checking for content changes since "${localContentUpdateTime}".`);

File: node_modules/gatsby-source-craft/gatsby-node.js:470:9



  Error: Failed to execute query LIST_Form.
  
  - paginate.ts:60 Object.paginate
    [panda]/[gatsby-graphql-source-toolkit]/src/source-nodes/fetch-nodes/paginate.ts:60:13
  
  - task_queues.js:97 processTicksAndRejections
    internal/process/task_queues.js:97:5
  
  - fetch-lists.ts:61 fetchNodeList
    [panda]/[gatsby-graphql-source-toolkit]/src/source-nodes/fetch-nodes/fetch-lists.ts:61:20
  
  - fetch-lists.ts:36 Object.fetchAllNodes
    [panda]/[gatsby-graphql-source-toolkit]/src/source-nodes/fetch-nodes/fetch-lists.ts:36:24
  
  - create-nodes.ts:13 Object.createNodes
    [panda]/[gatsby-graphql-source-toolkit]/src/source-nodes/node-actions/create-nodes.ts:13:20
  
  - async Promise.all
  
  - source-all-nodes.ts:20 sourceAllNodes
    [panda]/[gatsby-graphql-source-toolkit]/src/source-nodes/source-all-nodes.ts:20:3
  
  - gatsby-node.js:470 Object.exports.sourceNodes
    [panda]/[gatsby-source-craft]/gatsby-node.js:470:9
  
  - api-runner-node.js:485 runAPI
    [panda]/[gatsby]/src/utils/api-runner-node.js:485:16
  


```

The above is without using a custom fragment as the plugin indicates.
However if I use a custom fragment as indicated by gatsby-source-craft it works.

This is the custom [Fragment](https://gist.github.com/danieladarve/55038314f896172a4d9898d5778469d7) i am using

I made a parallel test using NextJS just for the fun of it and Formie is sourced without any issues.

**Comment by engram-design** [comment]
Thanks for the info @danieladarve I'll tee up a chat with @andris-sevcenko just to check this is all correct behaviour.

**Comment by mmorkt** [comment]
Just tried installing Formie on my Gatsby+Craft CMS site and now it won’t build. I’m getting the same error after installing the plugin

**Comment by mmorkt** [comment]
just an update. Removed the plugin and i can build again

**Comment by engram-design** [comment]
Fixed for the next release. To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.20",
  "...": "..."
}
```

Then run `composer update`. Thanks to @andris-sevcenko for pointing me in the right direction.

**Comment by engram-design** [comment]
Fixed in [1.4.21](https://github.com/verbb/formie/releases/tag/1.4.21)
